File: c++/053.cpp
class Solution {
//动态规划：dp[i] = max(dp[i-1]+nums[i],nums[i]); 结果为 dp 最大值
public:
    int maxSubArray(vector<int>& nums) {
        int n=nums.size();
        int dp[n];
        dp[0]=nums[0];
        for(int i=1;i<n;i++)
            dp[i] = max(dp[i-1]+nums[i],nums[i]);
        return *max_element(dp,dp+n);
    }
};

File: go/053.go
package main
/**
从整数数组中找出一个 具有最大和的连续子数组 返回其最大和
动态规划：f[i] 0-i的最大值
f[i-1] < 0 f[i] = n[i]
f[i-1] > 0 f[i] = f[i-1] + n[i]
f[0] = n[0]
*/
func maxSubArray(nums []int) int {
	var f = make([]int, len(nums))
	f[0] = nums[0]
	for i := 1; i < len(nums); i++ {
		if f[i-1] > 0 {
			f[i] = f[i-1] + nums[i]
		} else {
			f[i] = nums[i]
		}
	}
	var m = -10001
	for _, v := range f {
		m = max(m,v)
	}
	return m
}